tests/Fixtures/Integration/priority/escape_implicit_backslashes,single_quote.test

```diff
@@ -0,0 +1,11 @@
+--TEST--
+Integration of fixers: escape_implicit_backslashes,single_quote.
+--RULESET--
+{"escape_implicit_backslashes": {"double_quoted": true}, "single_quote": true}
```

Comment:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3155#discussion_r151743010

Reply:
ping @Slamdunk 
if I see correctly, it's the last raised concern

Reply:
I think the extended description of https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3155#issuecomment-347001043 can be enough: one single point to retrieve all the infos about how and why doing stuffs.

Reply:
yet, in this integration test, guy asked to not fix single quotes. and ultimatelly, they got fixed.

if in this confix would be a request to fix single_quoted strings as well, then output would be great. yet there wasn't.

Reply:
I think one of us is not getting the whole picture.

1. SingleQuoteFixer (which is not part of this PR) will always fix well-formatted double-quoted strings into single-quoted strings
1. EscapeImplicitBackslashesFixer, when configured with `double_quoted` enabled, will always fix non-well-formatted double-quoted strings to well-formatted double-quoted strings

So if we run SingleQuoteFixer before EscapeImplicitBackslashesFixer we would simply end up that, in a *second* run, SingleQuoteFixer will fix what EscapeImplicitBackslashesFixer fixed in the first run.

Hence, as they are logically consecutive, EscapeImplicitBackslashesFixer must run before SingleQuoteFixer.

> yet, in this integration test, guy asked to not fix single quotes. and ultimatelly, they got fixed.

No, single-quoted didn't get fixed. A double-quoted string got fixed into a single-quoted string by SingleQuoteFixer.

Take this example:

```php
$a = "foo \ bar";
```

With EscapeImplicitBackslashesFixer configured with `double_quoted: true` (and no matter whether EscapeImplicitBackslashesFixer's `single_quoted` config is false or true) will fix it into:

```php
$a = "foo \\ bar";
```

Only then, SingleQuoteFixer find the double-quoted string, sees it as well-formatted (all the escapes are even) and so it can move it into a single-quoted string:

```php
$a = 'foo \\ bar';
```

Reply:
OK, now I get it. thanks for explanation and patience.